### ISSUE TITLE
[BUG] Vignette building on macOS fails without helpful logs

### ISSUE THREAD

**Issue by flying-sheep** [lcolladotor/biocthis#27]
## Context

I’m using the biocthis GH action basically unchanged, and if fails on macOS while trying to build the vignettes, with nothing but the unhelpful and extremely terse message

```
Error: processing vignette 'Gene-Relevance.Rmd' failed with diagnostics:
polygon edge not found
```

Since the workflow you generate is extremely complicated, and the error log isn’t helpful, I can’t debug this.

Workflow definition: https://github.com/theislab/destiny/blob/91db4509ce9daaedbcb99d9bc8f5b5e5f63247f2/.github/workflows/check-bioc.yml

## Code

IDK what `--no-vignettes` in your workflow is supposed to mean, as `rcmdcheck::rcmdcheck()` clearly still tries to build the vignettes.

Unfortunately it does so without showing any warnings or tracebacks.

Where are the vignettes *supposed to* be built? How do I see the full logs when building them fails?

```R
options(crayon.enabled = TRUE)
rcmdcheck::rcmdcheck(
    args = c("--no-manual", "--no-vignettes", "--timings"),
    build_args = c("--no-manual", "--keep-empty-dirs", "--no-resave-data"),
    error_on = "warning",
    check_dir = "check"
)
```

```
── R CMD build ─────────────────────────────────────────────────────────────────
* checking for file ‘.../DESCRIPTION’ ... OK
* preparing ‘destiny’:
* checking DESCRIPTION meta-information ... OK
* cleaning src
* installing the package to build vignettes
* creating vignettes ... ERROR
Error: --- re-building ‘Diffusion-Map-recap.Rmd’ using rmarkdown
--- finished re-building ‘Diffusion-Map-recap.Rmd’
--- re-building ‘Diffusion-Maps.Rmd’ using rmarkdown
--- finished re-building ‘Diffusion-Maps.Rmd’
--- re-building ‘DPT.Rmd’ using rmarkdown
--- finished re-building ‘DPT.Rmd’
--- re-building ‘Gene-Relevance.Rmd’ using rmarkdown
Quitting from lines 119-121 (Gene-Relevance.Rmd) 
Error: Error: processing vignette 'Gene-Relevance.Rmd' failed with diagnostics:
polygon edge not found
--- failed re-building ‘Gene-Relevance.Rmd’
--- re-building ‘Global-Sigma.Rmd’ using rmarkdown
Quitting from lines 38-48 (Global-Sigma.Rmd) 
Error: Error: processing vignette 'Global-Sigma.Rmd' failed with diagnostics:
polygon edge not found
--- failed re-building ‘Global-Sigma.Rmd’
--- re-building ‘tidyverse.Rmd’ using rmarkdown
Quitting from lines 66-68 (tidyverse.Rmd) 
Error: Error: processing vignette 'tidyverse.Rmd' failed with diagnostics:
polygon edge not found
--- failed re-building ‘tidyverse.Rmd’
SUMMARY: processing the following files failed:
  ‘Gene-Relevance.Rmd’ ‘Global-Sigma.Rmd’ ‘tidyverse.Rmd’
Error: Error: Vignette re-building failed.
Execution halted
Error: Error in proc$get_built_file() : Build process failed
Calls: <Anonymous> ... build_package -> with_envvar -> force -> <Anonymous>
Execution halted
Error: Process completed with exit code 1.
```

## Small reproducible example

I’ll make one once you helped me get a proper log from the vignette building process.

## R session information

<details>
<summary>Session info</summary>

```
─ Session info ───────────────────────────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.1.2 (2021-11-01)
 os       Ubuntu 20.04.3 LTS
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       UTC
 date     2022-01-20
 pandoc   2.14.0.3 @ /usr/local/bin/pandoc

─ Packages ───────────────────────────────────────────────────────────────────────────────────────
 package                * version    date (UTC) lib source
 abind                    1.4-5      2016-07-21 [1] RSPM (R 4.1.0)
 AnnotationDbi            1.56.2     2021-11-09 [1] Bioconductor
 AnnotationFilter         1.18.0     2021-10-26 [1] Bioconductor
 AnnotationHub            3.2.0      2021-10-26 [1] Bioconductor
 AnVIL                    1.6.2      2021-12-26 [2] Bioconductor
 askpass                  1.1        2019-01-13 [2] RSPM (R 4.1.0)
 assertthat               0.2.1      2019-03-21 [1] RSPM (R 4.1.0)
 backports                1.4.1      2021-12-13 [1] RSPM (R 4.1.0)
 base                   * 4.1.2      2021-11-01 [3] local
 base64enc                0.1-3      2015-07-28 [1] RSPM (R 4.1.0)
 beachmat                 2.10.0     2021-10-26 [1] Bioconductor
 BH                       1.78.0-0   2021-12-15 [1] RSPM (R 4.1.0)
 Biobase                  2.54.0     2021-10-26 [1] Bioconductor
 BiocCheck                1.30.0     2021-10-26 [1] Bioconductor
 BiocFileCache            2.2.0      2021-10-26 [1] Bioconductor
 BiocGenerics             0.40.0     2021-10-26 [1] Bioconductor
 BiocIO                   1.4.0      2021-10-26 [1] Bioconductor
 BiocManager              1.30.16    2021-06-15 [2] CRAN (R 4.1.2)
 BiocNeighbors            1.12.0     2021-10-26 [1] Bioconductor
 BiocParallel             1.28.3     2021-12-09 [1] Bioconductor
 BiocSingular             1.10.0     2021-10-26 [1] Bioconductor
 BiocVersion              3.14.0     2021-05-19 [2] Bioconductor
 biocViews                1.62.1     2021-11-02 [1] Bioconductor
 biomaRt                  2.50.2     2022-01-13 [1] Bioconductor
 Biostrings               2.62.0     2021-10-26 [1] Bioconductor
 bit                      4.0.4      2020-08-04 [1] RSPM (R 4.1.0)
 bit64                    4.0.5      2020-08-30 [1] RSPM (R 4.1.0)
 bitops                   1.0-7      2021-04-24 [1] RSPM (R 4.1.0)
 blob                     1.2.2      2021-07-23 [1] RSPM (R 4.1.0)
 bluster                  1.4.0      2021-10-26 [1] Bioconductor
 boot                     1.3-28     2021-05-03 [3] CRAN (R 4.1.2)
 brew                     1.0-6      2011-04-13 [2] RSPM (R 4.1.0)
 brio                     1.1.3      2021-11-30 [2] RSPM (R 4.1.0)
 broom                    0.7.11     2022-01-03 [1] RSPM (R 4.1.0)
 bslib                    0.3.1      2021-10-06 [1] RSPM (R 4.1.0)
 cachem                   1.0.6      2021-08-19 [2] RSPM (R 4.1.0)
 callr                    3.7.0      2021-04-20 [2] RSPM (R 4.1.0)
 car                      3.0-12     2021-11-06 [1] RSPM (R 4.1.0)
 carData                  3.0-5      2022-01-06 [1] RSPM (R 4.1.0)
 caret                    6.0-90     2021-10-09 [1] RSPM (R 4.1.0)
 cellranger               1.1.0      2016-07-27 [1] RSPM (R 4.1.0)
 class                    7.3-20     2022-01-13 [3] RSPM (R 4.1.0)
 cli                      3.1.0      2021-10-27 [2] RSPM (R 4.1.0)
 clipr                    0.7.1      2020-10-08 [2] RSPM (R 4.1.0)
 cluster                  2.1.2      2021-04-17 [3] CRAN (R 4.1.2)
 codetools                0.2-18     2020-11-04 [3] CRAN (R 4.1.2)
 colorspace               2.0-2      2021-06-24 [1] RSPM (R 4.1.0)
 commonmark               1.7        2018-12-01 [2] RSPM (R 4.1.0)
 compiler                 4.1.2      2021-11-01 [3] local
 conflicted               1.1.0      2021-11-26 [1] RSPM (R 4.1.0)
 conquer                  1.2.1      2021-11-01 [1] RSPM (R 4.1.0)
 cowplot                  1.1.1      2020-12-30 [1] RSPM (R 4.1.0)
 cpp11                    0.4.2      2021-11-30 [2] RSPM (R 4.1.0)
 crayon                   1.4.2      2021-10-29 [2] RSPM (R 4.1.0)
 credentials              1.3.2      2021-11-29 [2] RSPM (R 4.1.0)
 crosstalk                1.2.0      2021-11-04 [1] RSPM (R 4.1.0)
 curl                     4.3.2      2021-06-23 [2] RSPM (R 4.1.0)
 data.table               1.14.2     2021-09-27 [1] RSPM (R 4.1.0)
 datasets               * 4.1.2      2021-11-01 [3] local
 DBI                      1.1.2      2021-12-20 [1] RSPM (R 4.1.0)
 dbplyr                   2.1.1      2021-04-06 [1] RSPM (R 4.1.0)
 DelayedArray             0.20.0     2021-10-26 [1] Bioconductor
 DelayedMatrixStats       1.16.0     2021-10-26 [1] Bioconductor
 DEoptimR                 1.0-10     2022-01-03 [1] RSPM (R 4.1.0)
 desc                     1.4.0      2021-09-28 [2] RSPM (R 4.1.0)
 destiny                  3.8.0      2022-01-20 [1] local (/__w/destiny/destiny)
 devtools                 2.4.3      2021-11-30 [2] RSPM (R 4.1.0)
 diffobj                  0.3.5      2021-10-05 [2] RSPM (R 4.1.0)
 digest                   0.6.29     2021-12-01 [2] RSPM (R 4.1.0)
 docopt                   0.7.1      2020-06-24 [2] RSPM (R 4.1.2)
 dplyr                    1.0.7      2021-06-18 [2] RSPM (R 4.1.0)
 dqrng                    0.3.0      2021-05-01 [1] RSPM (R 4.1.0)
 DT                       0.20       2021-11-15 [1] RSPM (R 4.1.0)
 dtplyr                   1.2.0      2021-12-05 [1] RSPM (R 4.1.0)
 e1071                    1.7-9      2021-09-16 [1] RSPM (R 4.1.0)
 edgeR                    3.36.0     2021-10-26 [1] Bioconductor
 ellipsis                 0.3.2      2021-04-29 [2] RSPM (R 4.1.0)
 ensembldb                2.18.3     2022-01-13 [1] Bioconductor
 evaluate                 0.14       2019-05-28 [2] RSPM (R 4.1.0)
 ExperimentHub            2.2.0      2021-10-26 [1] Bioconductor
 fansi                    1.0.2      2022-01-14 [2] RSPM (R 4.1.0)
 farver                   2.1.0      2021-02-28 [1] RSPM (R 4.1.0)
 fastmap                  1.1.0      2021-01-25 [2] RSPM (R 4.1.0)
 filelock                 1.0.2      2018-10-05 [1] RSPM (R 4.1.0)
 FNN                      1.1.3      2019-02-15 [1] RSPM (R 4.1.0)
 fontawesome              0.2.2      2021-07-02 [1] RSPM (R 4.1.0)
 forcats                  0.5.1      2021-01-27 [1] RSPM (R 4.1.0)
 foreach                  1.5.1      2020-10-15 [1] RSPM (R 4.1.0)
 foreign                  0.8-82     2022-01-13 [3] RSPM (R 4.1.0)
 formatR                  1.11       2021-06-01 [2] RSPM (R 4.1.0)
 fs                       1.5.2      2021-12-08 [2] RSPM (R 4.1.0)
 futile.logger            1.4.3      2016-07-10 [2] RSPM (R 4.1.0)
 futile.options           1.0.1      2018-04-20 [2] RSPM (R 4.1.0)
 future                   1.23.0     2021-10-31 [1] RSPM (R 4.1.0)
 future.apply             1.8.1      2021-08-10 [1] RSPM (R 4.1.0)
 gargle                   1.2.0      2021-07-02 [1] RSPM (R 4.1.0)
 generics                 0.1.1      2021-10-25 [2] RSPM (R 4.1.0)
 GenomeInfoDb             1.30.0     2021-10-26 [1] Bioconductor
 GenomeInfoDbData         1.2.7      2022-01-20 [1] Bioconductor
 GenomicAlignments        1.30.0     2021-10-26 [1] Bioconductor
 GenomicFeatures          1.46.3     2021-12-30 [1] Bioconductor
 GenomicRanges            1.46.1     2021-11-18 [1] Bioconductor
 gert                     1.5.0      2022-01-03 [2] RSPM (R 4.1.0)
 getopt                   1.20.3     2019-03-22 [1] RSPM (R 4.1.0)
 ggplot.multistats        1.0.0      2019-10-28 [1] RSPM (R 4.1.0)
 ggplot2                  3.3.5      2021-06-25 [1] RSPM (R 4.1.0)
 ggthemes                 4.2.4      2021-01-20 [1] RSPM (R 4.1.0)
 gh                       1.3.0      2021-04-30 [2] RSPM (R 4.1.0)
 gitcreds                 0.1.1      2020-12-04 [2] RSPM (R 4.1.0)
 globals                  0.14.0     2020-11-22 [1] RSPM (R 4.1.0)
 glue                     1.6.0      2021-12-17 [2] RSPM (R 4.1.0)
 googledrive              2.0.0      2021-07-08 [1] RSPM (R 4.1.0)
 googlesheets4            1.0.0      2021-07-21 [1] RSPM (R 4.1.0)
 gower                    0.2.2      2020-06-23 [1] RSPM (R 4.1.0)
 graph                    1.72.0     2021-10-26 [1] Bioconductor
 graphics               * 4.1.2      2021-11-01 [3] local
 grDevices              * 4.1.2      2021-11-01 [3] local
 grid                     4.1.2      2021-11-01 [3] local
 gridExtra                2.3        2017-09-09 [1] RSPM (R 4.1.0)
 gtable                   0.3.0      2019-03-25 [1] RSPM (R 4.1.0)
 haven                    2.4.3      2021-08-04 [1] RSPM (R 4.1.0)
 hexbin                   1.28.2     2021-01-08 [1] RSPM (R 4.1.0)
 highr                    0.9        2021-04-16 [2] RSPM (R 4.1.0)
 hms                      1.1.1      2021-09-26 [1] RSPM (R 4.1.0)
 htmltools                0.5.2      2021-08-25 [1] RSPM (R 4.1.0)
 htmlwidgets              1.5.4      2021-09-08 [1] RSPM (R 4.1.0)
 httpuv                   1.6.5      2022-01-05 [1] RSPM (R 4.1.0)
 httr                     1.4.2      2020-07-20 [2] RSPM (R 4.1.0)
 ids                      1.0.1      2017-05-31 [1] RSPM (R 4.1.0)
 igraph                   1.2.11     2022-01-04 [1] RSPM (R 4.1.0)
 ini                      0.3.1      2018-05-20 [2] RSPM (R 4.1.0)
 interactiveDisplayBase   1.32.0     2021-10-26 [1] Bioconductor
 ipred                    0.9-12     2021-09-15 [1] RSPM (R 4.1.0)
 IRanges                  2.28.0     2021-10-26 [1] Bioconductor
 irlba                    2.3.5      2021-12-06 [1] RSPM (R 4.1.0)
 isoband                  0.2.5      2021-07-13 [1] RSPM (R 4.1.0)
 iterators                1.0.13     2020-10-15 [1] RSPM (R 4.1.0)
 jquerylib                0.1.4      2021-04-26 [1] RSPM (R 4.1.0)
 jsonlite                 1.7.3      2022-01-17 [2] RSPM (R 4.1.0)
 KEGGREST                 1.34.0     2021-10-26 [1] Bioconductor
 KernSmooth               2.23-20    2021-05-03 [3] CRAN (R 4.1.2)
 knitr                    1.37       2021-12-16 [2] RSPM (R 4.1.0)
 knn.covertree            1.0        2019-10-28 [1] RSPM (R 4.1.0)
 labeling                 0.4.2      2020-10-20 [1] RSPM (R 4.1.0)
 laeken                   0.5.2      2021-10-06 [1] RSPM (R 4.1.0)
 lambda.r                 1.2.4      2019-09-18 [2] RSPM (R 4.1.0)
 later                    1.3.0      2021-08-18 [1] RSPM (R 4.1.0)
 lattice                  0.20-45    2021-09-22 [3] CRAN (R 4.1.2)
 lava                     1.6.10     2021-09-02 [1] RSPM (R 4.1.0)
 lazyeval                 0.2.2      2019-03-15 [1] RSPM (R 4.1.0)
 lifecycle                1.0.1      2021-09-24 [2] RSPM (R 4.1.0)
 limma                    3.50.0     2021-10-26 [1] Bioconductor
 listenv                  0.8.0      2019-12-05 [1] RSPM (R 4.1.0)
 littler                  0.3.15     2021-12-03 [2] RSPM (R 4.1.0)
 lme4                     1.1-27.1   2021-06-22 [1] RSPM (R 4.1.0)
 lmtest                   0.9-39     2021-11-07 [1] RSPM (R 4.1.0)
 locfit                   1.5-9.4    2020-03-25 [1] RSPM (R 4.1.0)
 lubridate                1.8.0      2021-10-07 [1] RSPM (R 4.1.0)
 magrittr                 2.0.1      2020-11-17 [2] RSPM (R 4.1.0)
 maptools                 1.1-2      2021-09-07 [1] RSPM (R 4.1.0)
 MASS                     7.3-55     2022-01-13 [3] RSPM (R 4.1.0)
 Matrix                   1.4-0      2021-12-08 [3] RSPM (R 4.1.0)
 MatrixGenerics           1.6.0      2021-10-26 [1] Bioconductor
 MatrixModels             0.5-0      2021-03-02 [1] RSPM (R 4.1.0)
 matrixStats              0.61.0     2021-09-17 [1] RSPM (R 4.1.0)
 memoise                  2.0.1      2021-11-26 [2] RSPM (R 4.1.0)
 metapod                  1.2.0      2021-10-26 [1] Bioconductor
 methods                * 4.1.2      2021-11-01 [3] local
 mgcv                     1.8-38     2021-10-06 [3] CRAN (R 4.1.2)
 mime                     0.12       2021-09-28 [2] RSPM (R 4.1.0)
 minqa                    1.2.4      2014-10-09 [1] RSPM (R 4.1.0)
 ModelMetrics             1.2.2.2    2020-03-17 [1] RSPM (R 4.1.0)
 modelr                   0.1.8      2020-05-19 [1] RSPM (R 4.1.0)
 munsell                  0.5.0      2018-06-12 [1] RSPM (R 4.1.0)
 nlme                     3.1-155    2022-01-13 [3] RSPM (R 4.1.0)
 nloptr                   1.2.2.3    2021-11-02 [1] RSPM (R 4.1.0)
 nnet                     7.3-17     2022-01-13 [3] RSPM (R 4.1.0)
 numDeriv                 2016.8-1.1 2019-06-06 [1] RSPM (R 4.1.0)
 openssl                  1.4.6      2021-12-19 [2] RSPM (R 4.1.0)
 optparse                 1.7.1      2021-10-08 [1] RSPM (R 4.1.0)
 org.Mm.eg.db             3.14.0     2022-01-20 [1] Bioconductor
 parallel                 4.1.2      2021-11-01 [3] local
 parallelly               1.30.0     2021-12-17 [1] RSPM (R 4.1.0)
 pbkrtest                 0.5.1      2021-03-09 [1] RSPM (R 4.1.0)
 pcaMethods               1.86.0     2021-10-26 [1] Bioconductor
 pillar                   1.6.4      2021-10-18 [2] RSPM (R 4.1.0)
 pkgbuild                 1.3.1      2021-12-20 [2] RSPM (R 4.1.0)
 pkgconfig                2.0.3      2019-09-22 [2] RSPM (R 4.1.0)
 pkgload                  1.2.4      2021-11-30 [2] RSPM (R 4.1.0)
 plogr                    0.2.0      2018-03-25 [1] RSPM (R 4.1.0)
 plyr                     1.8.6      2020-03-03 [1] RSPM (R 4.1.0)
 png                      0.1-7      2013-12-03 [1] RSPM (R 4.1.0)
 praise                   1.0.0      2015-08-11 [2] RSPM (R 4.1.0)
 prettyunits              1.1.1      2020-01-24 [2] RSPM (R 4.1.0)
 pROC                     1.18.0     2021-09-03 [1] RSPM (R 4.1.0)
 processx                 3.5.2      2021-04-30 [2] RSPM (R 4.1.0)
 prodlim                  2019.11.13 2019-11-17 [1] RSPM (R 4.1.0)
 progress                 1.2.2      2019-05-16 [1] RSPM (R 4.1.0)
 progressr                0.10.0     2021-12-19 [1] RSPM (R 4.1.0)
 promises                 1.2.0.1    2021-02-11 [1] RSPM (R 4.1.0)
 ProtGenerics             1.26.0     2021-10-26 [1] Bioconductor
 proxy                    0.4-26     2021-06-07 [1] RSPM (R 4.1.0)
 ps                       1.6.0      2021-02-28 [2] RSPM (R 4.1.0)
 purrr                    0.3.4      2020-04-17 [2] RSPM (R 4.1.0)
 quantreg                 5.86       2021-06-06 [1] RSPM (R 4.1.0)
 R6                       2.5.1      2021-08-19 [2] RSPM (R 4.1.0)
 ranger                   0.13.1     2021-07-14 [1] RSPM (R 4.1.0)
 rapiclient               0.1.3      2020-01-17 [2] RSPM (R 4.1.0)
 rappdirs                 0.3.3      2021-01-31 [2] RSPM (R 4.1.0)
 RBGL                     1.70.0     2021-10-26 [1] Bioconductor
 rcmdcheck                1.4.0      2021-09-27 [1] RSPM (R 4.1.0)
 RColorBrewer             1.1-2      2014-12-07 [1] RSPM (R 4.1.0)
 Rcpp                     1.0.8      2022-01-13 [1] RSPM (R 4.1.0)
 RcppArmadillo            0.10.7.5.0 2021-12-17 [1] RSPM (R 4.1.0)
 RcppEigen                0.3.3.9.1  2020-12-17 [1] RSPM (R 4.1.0)
 RcppHNSW                 0.3.0      2020-09-06 [1] RSPM (R 4.1.0)
 RCurl                    1.98-1.5   2021-09-17 [1] RSPM (R 4.1.0)
 readr                    2.1.1      2021-11-30 [1] RSPM (R 4.1.0)
 readxl                   1.3.1      2019-03-13 [1] RSPM (R 4.1.0)
 recipes                  0.1.17     2021-09-27 [1] RSPM (R 4.1.0)
 rematch                  1.0.1      2016-04-21 [1] RSPM (R 4.1.0)
 rematch2                 2.1.2      2020-05-01 [2] RSPM (R 4.1.0)
 remotes                  2.4.2      2021-11-30 [1] RSPM (R 4.1.0)
 repr                     1.1.4      2022-01-04 [1] RSPM (R 4.1.0)
 reprex                   2.0.1      2021-08-05 [1] RSPM (R 4.1.0)
 reshape2                 1.4.4      2020-04-09 [1] RSPM (R 4.1.0)
 restfulr                 0.0.13     2017-08-06 [1] RSPM (R 4.1.0)
 rgl                      0.108.3    2021-11-21 [1] RSPM (R 4.1.0)
 Rhtslib                  1.26.0     2021-10-26 [1] Bioconductor
 rjson                    0.2.21     2022-01-09 [1] RSPM (R 4.1.0)
 rlang                    0.4.12     2021-10-18 [2] RSPM (R 4.1.0)
 rmarkdown                2.11       2021-09-14 [1] RSPM (R 4.1.0)
 robustbase               0.93-9     2021-09-27 [1] RSPM (R 4.1.0)
 roxygen2                 7.1.2      2021-09-08 [2] RSPM (R 4.1.0)
 rpart                    4.1-15     2019-04-12 [3] CRAN (R 4.1.2)
 rprojroot                2.0.2      2020-11-15 [2] RSPM (R 4.1.0)
 Rsamtools                2.10.0     2021-10-26 [1] Bioconductor
 RSpectra                 0.16-0     2019-12-01 [1] RSPM (R 4.1.0)
 RSQLite                  2.2.9      2021-12-06 [1] RSPM (R 4.1.0)
 rstudioapi               0.13       2020-11-12 [2] RSPM (R 4.1.0)
 rsvd                     1.0.5      2021-04-16 [1] RSPM (R 4.1.0)
 rtracklayer              1.54.0     2021-10-26 [1] Bioconductor
 RUnit                    0.4.32     2018-05-18 [1] RSPM (R 4.1.0)
 rversions                2.1.1      2021-05-31 [2] RSPM (R 4.1.0)
 rvest                    1.0.2      2021-10-16 [1] RSPM (R 4.1.0)
 S4Vectors                0.32.3     2021-11-21 [1] Bioconductor
 sass                     0.4.0      2021-05-12 [1] RSPM (R 4.1.0)
 ScaledMatrix             1.2.0      2021-10-26 [1] Bioconductor
 scales                   1.1.1      2020-05-11 [1] RSPM (R 4.1.0)
 scatterplot3d            0.3-41     2018-03-14 [1] RSPM (R 4.1.0)
 scran                    1.22.1     2021-11-14 [1] Bioconductor
 scRNAseq                 2.8.0      2021-10-30 [1] Bioconductor
 scuttle                  1.4.0      2021-10-26 [1] Bioconductor
 selectr                  0.4-2      2019-11-20 [1] RSPM (R 4.1.0)
 sessioninfo              1.2.2      2021-12-06 [2] RSPM (R 4.1.0)
 shiny                    1.7.1      2021-10-02 [1] RSPM (R 4.1.0)
 SingleCellExperiment     1.16.0     2021-10-26 [1] Bioconductor
 sitmo                    2.0.2      2021-10-13 [1] RSPM (R 4.1.0)
 smoother                 1.1        2015-04-16 [1] RSPM (R 4.1.0)
 snow                     0.4-4      2021-10-27 [1] RSPM (R 4.1.0)
 sourcetools              0.1.7      2018-04-25 [1] RSPM (R 4.1.0)
 sp                       1.4-6      2021-11-14 [1] RSPM (R 4.1.0)
 SparseM                  1.81       2021-02-18 [1] RSPM (R 4.1.0)
 sparseMatrixStats        1.6.0      2021-10-26 [1] Bioconductor
 spatial                  7.3-15     2022-01-13 [3] RSPM (R 4.1.0)
 splines                  4.1.2      2021-11-01 [3] local
 SQUAREM                  2021.1     2021-01-13 [1] RSPM (R 4.1.0)
 statmod                  1.4.36     2021-05-10 [1] RSPM (R 4.1.0)
 stats                  * 4.1.2      2021-11-01 [3] local
 stats4                   4.1.2      2021-11-01 [3] local
 stringdist               0.9.8      2021-09-09 [1] RSPM (R 4.1.0)
 stringi                  1.7.6      2021-11-29 [2] RSPM (R 4.1.0)
 stringr                  1.4.0      2019-02-10 [2] RSPM (R 4.1.0)
 SummarizedExperiment     1.24.0     2021-10-26 [1] Bioconductor
 survival                 3.2-13     2021-08-24 [3] CRAN (R 4.1.2)
 sys                      3.4        2020-07-23 [2] RSPM (R 4.1.0)
 tcltk                    4.1.2      2021-11-01 [3] local
 testthat                 3.1.1      2021-12-03 [2] RSPM (R 4.1.0)
 tibble                   3.1.6      2021-11-07 [2] RSPM (R 4.1.0)
 tidyr                    1.1.4      2021-09-27 [2] RSPM (R 4.1.0)
 tidyselect               1.1.1      2021-04-30 [2] RSPM (R 4.1.0)
 tidyverse                1.3.1      2021-04-15 [1] RSPM (R 4.1.0)
 timeDate                 3043.102   2018-02-21 [1] RSPM (R 4.1.0)
 tinytex                  0.36       2021-12-19 [1] RSPM (R 4.1.0)
 tools                    4.1.2      2021-11-01 [3] local
 TTR                      0.24.3     2021-12-12 [1] RSPM (R 4.1.0)
 tzdb                     0.2.0      2021-10-27 [1] RSPM (R 4.1.0)
 usethis                  2.1.5      2021-12-09 [2] RSPM (R 4.1.0)
 utf8                     1.2.2      2021-07-24 [2] RSPM (R 4.1.0)
 utils                  * 4.1.2      2021-11-01 [3] local
 uuid                     1.0-3      2021-11-01 [1] RSPM (R 4.1.0)
 vcd                      1.4-9      2021-10-18 [1] RSPM (R 4.1.0)
 vctrs                    0.3.8      2021-04-29 [2] RSPM (R 4.1.0)
 VIM                      6.1.1      2021-07-22 [1] RSPM (R 4.1.0)
 viridis                  0.6.2      2021-10-13 [1] RSPM (R 4.1.0)
 viridisLite              0.4.0      2021-04-13 [1] RSPM (R 4.1.0)
 vroom                    1.5.7      2021-11-30 [1] RSPM (R 4.1.0)
 waldo                    0.3.1      2021-09-14 [2] RSPM (R 4.1.0)
 whisker                  0.4        2019-08-28 [2] RSPM (R 4.1.0)
 withr                    2.4.3      2021-11-30 [2] RSPM (R 4.1.0)
 xfun                     0.29       2021-12-14 [2] RSPM (R 4.1.0)
 XML                      3.99-0.8   2021-09-17 [1] RSPM (R 4.1.0)
 xml2                     1.3.3      2021-11-30 [2] RSPM (R 4.1.0)
 xopen                    1.0.0      2018-09-17 [2] RSPM (R 4.1.0)
 xtable                   1.8-4      2019-04-21 [1] RSPM (R 4.1.0)
 xts                      0.12.1     2020-09-09 [1] RSPM (R 4.1.0)
 XVector                  0.34.0     2021-10-26 [1] Bioconductor
 yaml                     2.2.1      2020-02-01 [2] RSPM (R 4.1.0)
 zip                      2.2.0      2021-05-31 [2] RSPM (R 4.1.0)
 zlibbioc                 1.40.0     2021-10-26 [1] Bioconductor
 zoo                      1.8-9      2021-03-09 [1] RSPM (R 4.1.0)

 [1] /__w/_temp/Library
 [2] /usr/local/lib/R/site-library
 [3] /usr/local/lib/R/library

──────────────────────────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by lcolladotor** [comment]
Hi @flying-sheep,

I don't appreciate the tone of your message and it doesn't make me want to help you. You might benefit from checking these tweets https://gist.github.com/jimhester/9b5461108f93162e15c42684076de82f#file-tweets-md.


Having said that, in order to run `R CMD check` you first need a package tarball which is created (_built_) with `R CMD build`. The R package `rcmdcheck` helps do that from within R and is used by many people. You can find it at https://cran.r-project.org/web/packages/rcmdcheck/index.html. If we look at the help file for `rcmdcheck::rcmdcheck()` at https://r-lib.github.io/rcmdcheck/reference/rcmdcheck.html you'll see that `build_args` specifies options for `R CMD build` while `args` specifies options for `R CMD check`. Since we don't want `R CMD check` to re-make the vignettes, we turn that off there. This is analogous to what is done by Bioconductor at say at https://bioconductor.org/checkResults/release/bioc-LATEST/biocthis/nebbiolo2-buildsrc.html (`R CMD build`) and https://bioconductor.org/checkResults/release/bioc-LATEST/biocthis/nebbiolo2-checksrc.html (`R CMD check`).

The GitHub action workflow provided by `biocthis` simply helps you run `rcmdcheck::rcmdcheck()` in a borrowed computer and a Bioconductor-friendly environment. 


Going into the specifics of your error, from your log, you can see that there's an error at lines 119-221 from `Gene-Relevance.Rmd` (`Quitting from lines 119-121 (Gene-Relevance.Rmd) `), and similarly for other vignette files you have. So, you could try:

* Manually running the code for that Rmd up to and including those lines. See if you get an error.
* Try it again but in an environment that has the same packages and versions as those used in the borrowed machines. You can see the R session information at https://github.com/theislab/destiny/runs/4885941840?check_suite_focus=true#step:20:22.
* Since this issue only pops up on macOS, there might be a system dependency missing. For example, searching on Google `polygon edge not found rmarkdown macos` leads to https://github.com/tidyverse/ggplot2/issues/2252 among other threads. There you can see that @bschilder has tried a few things at https://github.com/NathanSkene/EWCE/commits/master such as https://github.com/NathanSkene/EWCE/commit/1b9c6f1cae6efb796e2956fe8fc1e5adf2c21189. 

You could also choose to ignore macOS by commenting out https://github.com/theislab/destiny/blob/91db4509ce9daaedbcb99d9bc8f5b5e5f63247f2/.github/workflows/check-bioc.yml#L58. 

Anyway, enjoy the ride down this rabbit hole.

Best,
Leo

**Comment by flying-sheep** [comment]
Hi @lcolladotor, I did write that report while being frustrated by Bioconductor’s leadership’s intransparent and arbitrary decision making forcing me to do useless churn work. My package and its vignettes worked fine for years until they basically told me “we’re not going to tell you why, but you must now use Rmd for your vignettes, no discussion” … So it seems that some of that frustration leaked into my tone, and for that I’m sorry, it’s not your fault.

> Since we don't want R CMD check to re-make the vignettes, we turn that off there.

Well, at least that’s the intention right? The command output I pasted above is from an invocation with that option set, and it still seems to build the vignettes, right?

> Since this issue only pops up on macOS, there might be a system dependency missing.

Yes, that’s my assumption as well. I filed this issue to get help with debugging this and in turn to help you update the system dependencies in your GH action template. I found the issue you linked (thanks!) which would only be a last ditch effort, since instead of fixing the underlying issue, it will just result in broken plots in the vignette, right?

> You could also choose to ignore macOS

Sure, but that would defeat the purpose of the unit tests. Since Bioconductor’s build bots only run every few days, it’s important to find possible errors early.

**Comment by lcolladotor** [comment]
Hi again,

Regarding `rcmdcheck::rcmdcheck()` and `R CMD build` and `R CMD check` in general, `R CMD build` compiles the vignettes. So you can see that we are under the `rcmdcheck::rcmdcheck()` output section related to `R CMD build` given the first line of your log:

```
── R CMD build ─────────────────────────────────────────────────────────────────
```

`R CMD check` is never run by `rcmdcheck::rcmdcheck()` in your case, because `R CMD build` fails to make the vignettes. On a case where things are working, once `R CMD build` is done, `R CMD check` would follow and that's where here in `biocthis` as well as in Bioconductor, we don't want to spend re-making the vignettes since we just compiled them in the previous step (`R CMD build`) using the exact same R version, environment and packages that we'll be using for running `R CMD check`. If we had run `R CMD build` and `R CMD check` in different computers/environments, then it would make sense to have `R CMD check` re-make the vignettes.

Regarding the macOS-specific `polygon edge not found`, let's see if someone else comes up with a solution for it. If the solution involves installing a system dependency, then I might need to update `biocthis`. If it's say, an issue that is addressed internally in the `grid` package, then I won't need to change `biocthis`. 

I see at https://bioconductor.org/checkResults/release/bioc-LATEST/destiny/ that on the Bioconductor release branch, your package `destiny` is ok though on macOS. 

As for switching vignette engine builders to `knitr` and `Rmd` files, I couldn't find such announcements on the `bioc-devel` mailing list, though I might have missed it. I see that Levi's team is interested in helping those who are changing their vignettes though at https://stat.ethz.ch/pipermail/bioc-devel/2022-January/018785.html. 

Best,
Leo

**Comment by flying-sheep** [comment]
Thank you for the  detailled explanation regarding vignette building in `build`/`check`, I understand now!

> Regarding the macOS-specific polygon edge not found, let's see if someone else comes up with a solution for it.

Sure, but if if it breaks on the bioconductor build bots (which can still happen, see below) I need to fix this now. And to be able to debug this now, I think I need to be able to see warnings raised during `R CMD build`. If there is a way to do this (e.g. some `verbose` flag for `R CMD build`, or using `--log` and then `cat`ing it) it would make sense to enable it in `biocthis`. That’s why I filed this issue: Because I’m thinking I’m not the only one who could benefit from a more verbose vignette building output here.

> I see at bioconductor.org/checkResults/release/bioc-LATEST/destiny that on the Bioconductor release branch, your package destiny is ok though on macOS.

At the time of writing it says “Snapshot date 2022-01-20” which means that it ran on code from 3 days ago. 3 days ago, the code still used my old vignette building system. From all that I can see (e.g. those green checkmarks) it works fine. Still, the Bioconductor people told me I have to switch away from that one to Rmd. And in that process I encountered the errors mentioned above, which I’m not able to debug without help.

**Comment by flying-sheep** [comment]
Hi @lcolladotor, since you marked this as completed: Could you please mention which action version (or if unreleased: PR) is this completed in? Thank you!

**Comment by lcolladotor** [comment]
Your `polygon edge not found` error is unrelated to `biocthis`. See https://stat.ethz.ch/pipermail/bioc-devel/2022-January/018795.html for more details. It's thus out of scope for this issue page. 

If you have questions about this error, post them on the [bioc-devel mailing list](https://stat.ethz.ch/mailman/listinfo/bioc-devel) where other Bioconductor contributors or Core members might be able to help you.

**Comment by flying-sheep** [comment]
If I understand correctly, `rcmdcheck()` will call `R CMD build`, but doesn’t set it up so it shows errors that occur during vignette building.

I filed this issue assuming that `rcmdcheck()` can be run in a way that makes that happen. So I’m confused why this issue is marked as “closed, reason: completed”:

- If my assumption is correct, `biocthis` should change its generated code so `rcmdcheck()` will show warnings that happen during vignette building. In this case, I’d expect this issue’s status to be “open”.
- If my assumption is wrong, we need to file an issue for rcmdcheck so this is doable. In that case, depending on your issue policy, I’d expect this issue to be “open” (waiting for rcmdcheck to fix things) or “closed, reason: not planned”.
